.circleci/README.md

```diff
@@ -0,0 +1,69 @@
+variables that you can define in the CircleCI project settings to customize the build.
+
+## Where do I define the variables?
+For each set of variables, some you can define in the config.yml, in the `environment`
+section for a workflow step, for example, here is setting the variable `TIMEZONE`
+to be defined as `/usr/share/zoneinfo/America/Los_Angeles` available during the
+"setup" step:
+
+```bash
+  setup:
+    environment:
+      - TIMEZONE: "/usr/share/zoneinfo/America/Los_Angeles"
+```
+
+or you can define variables in the [CircleCI project settings](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project). 
+The latter is recommended for sensitive information. 
+
+For all configuration descriptions below, **all variables** from both tables 
+can be set in just the CircleCI project settings, if desired.
+
+# Templates
+
+## config.yml
+
+This is a base, or default configuration to build.
+
+**Variables that must be set in CircleCI Project Settings**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`DOCKER_USER`|Docker username to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+|`DOCKER_PASS`|Docker password to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+
+**Variables that can be set in config.yml**
+ 
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`CONTAINER_NAME`|The name of the Docker Hub container to deploy|no|`<ORG>/<REPO>`|
+|`DOCKER_TAG`|A custom tag for the deployed image.|no|the first 10 characters of the commit.|
+
+
+## config.singularity
+
+This configuration is equivalent to the default config.yml, with the additional change in permissions of the
+repo2docker folder to allow for the user to write given export into a Singularity container. If you want your
+container notebooks to be usable on a shared resource, you should choose this configuration. Note that the
+save happens in a temporary location provided by the container, and so you would want to save your notebooks as an export to your local machine, or to another location on your shared resource.
+
+**Variables that must be set in CircleCI Project Settings**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`DOCKER_USER`|Docker username to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+|`DOCKER_PASS`|Docker password to deploy to Docker Hub. The build will not deploy without it.|no|unset|
+
+**Variables that can be set in config.yml**
+
+|Variable|Description|Required|Default|
+|---|---|---|---|
+|`CONTAINER_NAME`|The name of the Docker Hub container to deploy|no|`<ORG>/<REPO>`|
+|`DOCKER_TAG`|A custom tag for the deployed image.|no|the first 10 characters of the commit.|
+|`NOTEBOOK_PERMISSION`|The permission to (recursively) give the notebook, to allow write.|no|777|
+|`NOTEBOOK_USERNAME`|The username (to derive the home directory) to build the notebook and virtual environment. (e.g., `/home/joyvan`|no|`joyvan`|
```

Comment:
-> "The username used inside the container"

Reply:
In Singularity world, this is just the notebook username (the user remains him/herself when in use).

Reply:
What do you mean with the notebook username? I was assuming that inside the container there is a user that owns the files copied over from the repository and that this variable sets that username. (and that for singularity the username that owns the files has to match the username of the person executing the container)

Reply:
The container is going to be sitting on a shared HPC resource, and more than one user could use it. The username is the name of the folder, and the name of the --username arg given for the container. If it's not general (or assumes use by any user) the container doesn't have much purpose for reproducibility (e.g., sharing with others).